open_fdd/mcp_rag/mcp_adapter.py

```
@@ -0,0 +1,136 @@
+        line = raw.strip()
+        if not line:
+            continue
+        try:
+            req = json.loads(line)
+            if not isinstance(req, dict):
+                out = _jsonrpc_error(None, -32600, "Invalid Request")
+            else:
+                out = handle_jsonrpc_request(req, adapter)
+        except json.JSONDecodeError:
+            out = _jsonrpc_error(None, -32700, "Parse error")
+        except Exception as exc:  # pragma: no cover
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)